### ISSUE TITLE
Validator node not starting

### ISSUE THREAD

**Issue by wouterdegeus** [sigp/lighthouse#2204]
## Description

Was running v1.1.0 just fine (both Beacon and Validator nodes), built v1.1.1 and restarted. Beacon node comes up just fine, validator node gets stuck after `INFO Starting validator client` message.

Note that I'm building this using Docker with a somewhat customized Dockerfile to have it run as non-root user:
```
FROM rust:1.50.0 AS builder
RUN apt-get update && apt-get install -y cmake
COPY . lighthouse
ARG PORTABLE
ENV PORTABLE \$PORTABLE
RUN cd lighthouse && make && make install-lcli

FROM debian:buster-slim
RUN apt-get update && apt-get install -y --no-install-recommends \
  libssl-dev \
  ca-certificates \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/*
COPY --from=builder /usr/local/cargo/bin/lighthouse /usr/local/bin/lighthouse
COPY --from=builder /usr/local/cargo/bin/lcli /usr/local/bin/lcli
RUN adduser --disabled-password --gecos "" --home /home/lighthouse lighthouse \
  && chown lighthouse:lighthouse /home/lighthouse
USER lighthouse
WORKDIR /home/lighthouse
```

When starting the new version it begins with a warning and then stalls after the validator client notice.
I tried restarting the service a few times, same result.
```
thread 'tokio-runtime-worker' panicked at 'missing scheme', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/warp-0.3.0/src/filters/cors.rs:620:37
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace                                                                                              
Feb 16 16:54:19.689 INFO Lighthouse started                      version: Lighthouse/v1.1.1-f819ba5+
Feb 16 16:54:19.689 INFO Configured for network                  name: mainnet 
Feb 16 16:54:19.689 INFO Starting validator client               validator_dir: "/home/lighthouse/.lighthouse/mainnet/validators", beacon_nodes: ["http://lighthouse_beacon_node:5052"]                                                                                                                                                                
Feb 16 17:00:06.632 INFO Shutting down..            
```
```
thread 'tokio-runtime-worker' panicked at 'missing scheme', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/warp-0.3.0/src/filters/cors.rs:620:37
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace                                                                                              
Feb 16 17:00:08.396 INFO Lighthouse started                      version: Lighthouse/v1.1.1-f819ba5+                 
Feb 16 17:00:08.396 INFO Configured for network                  name: mainnet                                                                                             
Feb 16 17:00:08.396 INFO Starting validator client               validator_dir: "/home/lighthouse/.lighthouse/mainnet/validators", beacon_nodes: ["http://lighthouse_beacon_node:5052"]                                                                                                                                                                
Feb 16 17:02:14.833 INFO Shutting down..      
```
```                                                                                                                             
thread 'tokio-runtime-worker' panicked at 'missing scheme', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/warp-0.3.0/src/filters/cors.rs:620:37
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace                                                                                              
Feb 16 17:02:16.809 INFO Lighthouse started                      version: Lighthouse/v1.1.1-f819ba5+                 
Feb 16 17:02:16.809 INFO Configured for network                  name: mainnet                                                                                             
Feb 16 17:02:16.809 INFO Starting validator client               validator_dir: "/home/lighthouse/.lighthouse/mainnet/validators", beacon_nodes: ["http://lighthouse_beacon_node:5052"]                                                                                                                                                                
```

Note that the tokio panick is also shown in the Beacon node when starting, but that seems to chug along just fine:
```
Feb 16 16:53:47.557 INFO Lighthouse started                      version: Lighthouse/v1.1.1-f819ba5+                                                                                                                                                                                                                                                   
Feb 16 16:53:47.567 INFO Configured for network                  name: mainnet                                                                                                                                                                                                                                                                         
Feb 16 16:53:47.584 INFO Data directory initialised              datadir: /home/lighthouse/.lighthouse/mainnet                                                                                                                                                                                                                                         
Feb 16 16:53:47.585 WARN Running HTTP server on port 5052                                                                                                                                                                                                                                                                                              
Feb 16 16:53:47.591 INFO Deposit contract                        address: 0x00000000219ab540356cbb839cbe05303d7705fa, deploy_block: 11184524                                                                                                                                                                                                           
Feb 16 16:53:48.822 INFO Hot-Cold DB initialized                 split_state: 0x16f4d341fec531bc281762834b9792d268b4512ac744b5b7e44ae07f8dd08bf0, split_slot: 555488, version: 2, service: freezer_db                                                                                                                                                  
Feb 16 16:53:48.827 INFO Started monitoring validator            pubkey: <MYPUBKEY>, service: val_mon                                                                                                                                                          
Feb 16 16:53:48.836 INFO Starting beacon chain                   method: resume, service: beacon                                                                                                                                                                                                                                                       
Feb 16 16:53:55.744 INFO Block production enabled                method: json rpc via http, endpoints: ["http://openethereum:8545"]                                                                                                                                                                                                                    
Feb 16 16:53:56.153 INFO Beacon chain initialized                head_slot: 555566, head_block: 0xd98a…006e, head_state: 0x1841…35f4, service: beacon                                                                                                                                                                                                  
Feb 16 16:53:56.153 INFO Timer service started                   service: node_timer                                                                                                                                                                                                                                                                   
Feb 16 16:53:56.159 INFO UPnP Attempting to initialise routes    service: UPnP                                                                                                                                                                                                                                                                         
Feb 16 16:53:56.192 INFO Libp2p Service                          peer_id: <censored>, service: libp2p                                                                                                                                                                                                       
Feb 16 16:53:56.212 INFO ENR Initialised                         tcp: Some(9000), udp: None, ip: None, id: <censored>, seq: 941, enr: enr:-<censored>, service: libp2p                                                                                                                                                                                                                                                                                                                  
Feb 16 16:53:56.228 INFO Listening established                   address: /ip4/0.0.0.0/tcp/9000/p2p/<censored>, service: libp2p                                                                                                                                                                             
thread 'tokio-runtime-worker' panicked at 'missing scheme', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/warp-0.3.0/src/filters/cors.rs:620:37                                                                                                                                                                                            
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace                                                                                                                                                                                                                                                                          
Feb 16 16:53:56.456 INFO HTTP API started                        listen_address: 0.0.0.0:5052                                                                                                                                                                                                                                                          
Feb 16 16:53:56.804 INFO Address updated                         udp_port: 9000, ip: <censored>, service: libp2p                                                                                                                                                                                                                                   
Feb 16 16:53:57.065 INFO Sync state updated                      new_state: Synced, old_state: Stalled, service: sync                                                                                                                                                                                                                                  
Feb 16 16:53:57.065 INFO Subscribed to topics                    topics: [BeaconBlock, BeaconAggregateAndProof, VoluntaryExit, ProposerSlashing, AttesterSlashing], service: network                                                                                                                                                                   
```

Kept the Beacon node at v1.1.1 and reverted the Validator to v1.1.0 and they're chugging along just fine again.
v1.1.0 starts with no errors:
```
Feb 16 17:02:30.858 INFO Lighthouse started                      version: Lighthouse/v1.1.0-e4b6213+
Feb 16 17:02:30.880 INFO Configured for network                  name: mainnet
Feb 16 17:02:30.880 INFO Starting validator client               validator_dir: "/home/lighthouse/.lighthouse/mainnet/validators", beacon_nodes: ["http://lighthouse_beacon_node:5052"]
Feb 16 17:02:30.880 INFO Metrics HTTP server started             listen_address: 0.0.0.0:5064
Feb 16 17:02:30.920 INFO Completed validator discovery           new_validators: 0
Feb 16 17:02:32.970 INFO Enabled validator                       voting_pubkey: <censored>
Feb 16 17:02:32.970 INFO Initialized validators                  enabled: 1, disabled: 0
Feb 16 17:02:33.175 INFO Connected to beacon node                endpoint: http://lighthouse_beacon_node:5052/, version: Lighthouse/v1.1.1-f819ba5+/x86_64-linux
```

## Version

Lighthouse v1.1.1
Rust v150

## Present Behaviour

Validator client seems stuck.

## Expected Behaviour

Start validating blocks like it did in v1.1.0

## Steps to resolve

Unknown.

**Comment by realbigsean** [comment]
@wouterdegeus could you share the commands you are using to start up the BN and VC? I think this might be due to an issue parsing a url associated with the `--http-allow-origin` flag.

this looks like where the panic is coming from: https://github.com/seanmonstar/warp/blob/29651e9adaff8ced761b0a0e61e1e729abe18271/src/filters/cors.rs#L619

**Comment by wouterdegeus** [comment]
Using docker-compose, command for the beacon node:
```
    command: lighthouse --network mainnet bn --staking --http --http-address 0.0.0.0 --metrics --metrics-allow-origin '*' --metrics-address 0.0.0.0 --eth1-endpoints http://openethereum:8545 --validator-monitor-pubkeys ${VALIDATORPUBS}
```
Validator:
```
    command: lighthouse --network mainnet vc --beacon-nodes http://lighthouse_beacon_node:5052  --metrics --metrics-allow-origin '*' --metrics-address 0.0.0.0 --graffiti ${GRAFFITI}
```
The graffiti tag comes from my .env file.

**Comment by realbigsean** [comment]
Thanks!

I'd try removing `--metrics-allow-origin '*'` from both commands, and see if that fixes things. It might mess up your ability gather metrics for the time being, but I'll start looking into a fix

**Comment by wouterdegeus** [comment]
That fixes it - node is now up and running on v1.1.1 without warnings.
Appears like my prometheus can access it without the allow-origin specification, so this is probably more secure anyway. I've been tinkering with the monitoring config last month :)
Thanks!

**Comment by paulhauner** [comment]
Thanks for raising this thoughtful and detailed issue @wouterdegeus :)

A fix has been released in https://github.com/sigp/lighthouse/releases/tag/v1.1.2
